pkg/kubectl/cmd/exec.go

```diff
@@ -161,6 +162,10 @@ func (p *ExecOptions) Complete(f *cmdutil.Factory, cmd *cobra.Command, argsIn []
 		}
 	}
 
+	if strings.HasPrefix(p.PodName, "pod/") {
```

Comment:
Better yet switch the command to use the resource builder instead of a bare client and use [ResourceNames](https://github.com/kubernetes/kubernetes/blob/47e3d23a67c371143bc84a2662f0f7bf410c3921/pkg/kubectl/cmd/logs.go#L170). Look at how `kubectl logs` is implemented.